Library to package

dglaude opened the pull request:
This split the single file library into a multi-file package in a folder.
That way I2C version and UART version can be included separately while sharing some code in __init__.

dglaude committed:
Moving file into folder as __init__ and take two copies

dglaude committed:
Add duplication of __init__.py (history lost)

dglaude committed:
Splitting into file per connectivity

dglaude committed:
Separate simpletest for uart and i2c + fix time import in I2C

dglaude committed:
Black cleaning

dglaude committed:
Pylint on import and what should be where

dglaude committed:
Merge back I2C and UART simpletest to avoid changing the docs

dglaude commented:
It seems to pass all the test. This is a fix for #5 that limit memory usage in any case.
Please consider this better than #7

brentru commented on examples/pm25_simpletest.py:
```diff
@@ -1,13 +1,15 @@
-import adafruit_pm25
+from adafruit_pm25.i2c import PM25_I2C
+
+# from adafruit_pm25.uart import PM25_UART
```
You may want to move this import under the `# Connect to a PM2.5 sensor over UART` or say "uncomment this if you are using UART`

kattni replied:
Please move this under `# Connect to a PM2.5 sensor over UART`. I believe that is a better place for it, and it will be more obvious that it should be uncommented for UART.

dglaude replied:
Done, but I start to wonder if it would not be best to have two separate "simpletest", one for I2C and one for UART. Just to keep think more simple.

brentru commented on adafruit_pm25/uart.py:
```diff
@@ -0,0 +1,85 @@
+        while True:
+            b = self._uart.read(1)
+            if not b:
+                print(b)  ### <<< debug
```
Don't `print` in the library

dglaude replied:
Done

brentru reviewed (changes requested):
This looks good - thanks for adding an extra protocol - requesting two small nit-picky changes

kattni commented on adafruit_pm25/__init__.py:
```diff
@@ -44,10 +44,7 @@
 """
 
 # imports
```
Please delete the `# imports` comment from this line.

kattni commented on adafruit_pm25/i2c.py:
```diff
@@ -0,0 +1,84 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_pm25`
```
This needs to be:
```suggestion
`adafruit_pm25.i2c`
```

kattni commented on adafruit_pm25/i2c.py:
```diff
@@ -0,0 +1,84 @@
+`adafruit_pm25`
+================================================================================
+
+CircuitPython library for PM2.5 Air Quality Sensors
```
Please update this line to reflect it being the i2c module for the library, such as:
```suggestion
I2C module for CircuitPython library for PM2.5 Air Quality Sensors
```

kattni commented on adafruit_pm25/i2c.py:
```diff
@@ -0,0 +1,84 @@
+
+**Hardware:**
+
+Works with most (any?) Plantower UART or I2C interfaced PM2.5 sensor.
```
This should not mention UART as it is now an I2C module.

kattni commented on adafruit_pm25/i2c.py:
```diff
@@ -0,0 +1,84 @@
+
+class PM25_I2C(PM25):
+    """
+    A driver for the PM2.5 Air quality sensor over I2C
```
Please update to:
```suggestion
    A module for using the PM2.5 Air quality sensor over I2C
```

kattni commented on adafruit_pm25/uart.py:
```diff
@@ -0,0 +1,85 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_pm25`
```
This should be:
```suggestion
`adafruit_pm25.uart`
```

kattni commented on adafruit_pm25/uart.py:
```diff
@@ -0,0 +1,85 @@
+`adafruit_pm25`
+================================================================================
+
+CircuitPython library for PM2.5 Air Quality Sensors
```
Please update this to reflect being the UART module, such as:
```suggestion
UART module for CircuitPython library for PM2.5 Air Quality Sensors
```

kattni commented on adafruit_pm25/uart.py:
```diff
@@ -0,0 +1,85 @@
+
+**Hardware:**
+
+Works with most (any?) Plantower UART or I2C interfaced PM2.5 sensor.
```
Should not mention I2C as this is the UART module.

kattni commented on adafruit_pm25/uart.py:
```diff
@@ -0,0 +1,85 @@
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+* Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
```
Is Bus Device necessary for UART?

dglaude replied:
Indeed, that what the whole point of this separation.

kattni commented on adafruit_pm25/uart.py:
```diff
@@ -0,0 +1,85 @@
+
+"""
+
+# imports
```
Please remove this comment.

kattni commented on adafruit_pm25/i2c.py:
```diff
@@ -0,0 +1,84 @@
+
+"""
+
+# imports
```
Please remove this comment.

kattni commented on examples/pm25_simpletest.py:
```diff
@@ -1,13 +1,15 @@
 """
-Example sketch to connect to PM2.5 sensor with either I2C or UART.
+Example sketch to connect to PM2.5 sensor with I2C.
```
It appears you include the UART code commented out. Why did you change the docstring to remove UART?

dglaude replied:
restaured

kattni reviewed (changes requested):
@dglaude This looks really good. There are some necessary changes for Sphinx to build properly, noted below. 

You also need to update this file: https://github.com/adafruit/Adafruit_CircuitPython_PM25/blob/master/docs/api.rst to look more like this for example: https://github.com/adafruit/Adafruit_CircuitPython_CircuitPlayground/blob/main/docs/api.rst

Without updating that, Sphinx will not render it properly. I may have missed something, we'll find out if the checks fail after you make the listed changes.

Thanks!

dglaude committed:
Update adafruit_pm25/i2c.py

Co-authored-by: Kattni <kattni@adafruit.com>

dglaude committed:
Update adafruit_pm25/i2c.py

Co-authored-by: Kattni <kattni@adafruit.com>

dglaude committed:
Update adafruit_pm25/i2c.py

Co-authored-by: Kattni <kattni@adafruit.com>

dglaude committed:
Update adafruit_pm25/uart.py

Co-authored-by: Kattni <kattni@adafruit.com>

dglaude committed:
Update adafruit_pm25/uart.py

Co-authored-by: Kattni <kattni@adafruit.com>

dglaude committed:
Update uart.py

Does not require busio, that was the whole point of separating.

dglaude committed:
Update uart.py

Remove print/debug code.

dglaude committed:
Update pm25_simpletest.py

Moving UART import near the part about UART.

dglaude committed:
Update __init__.py

dglaude committed:
Update __init__.py

dglaude committed:
Update i2c.py

dglaude committed:
Update uart.py

dglaude committed:
Update pm25_simpletest.py

dglaude committed:
Update api.rst

List all the module

dglaude commented:
I think I resolved all the change request, including api.rst.
The only thing I would like to argue is that there should be two example rather than one, an I2C version and an UART version.

Should I rename and simplify the current by remove all the UART comment, and make an UART only version?

dglaude committed:
Update setup.py

This might make PyPi more happy (as seen in Adafruit_CircuitPython_LSM6DS).

dglaude committed:
Temporary fix for pylint 2.6.0 upgrade on GitHub

New version of pylint (2.6.0) on GitHub raise this kind of message:
adafruit_pm25/i2c.py:84:16: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)

dglaude committed:
Adapting to W0707 pylint message

W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from):
> 2020-08-23T08:28:02.9225541Z ************* Module adafruit_pm25.i2c
> 2020-08-23T08:28:02.9226318Z adafruit_pm25/i2c.py:84:16: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)

dglaude committed:
Undo hiding the pylint issue

This was hiding W0707: "Consider explicitly re-raising using the 'from' keyword (raise-missing-from)"

dglaude commented:
Maybe those two learn guide will need attention once this is approuved:

- https://learn.adafruit.com/pm25-air-quality-sensor/python-and-circuitpython
- https://learn.adafruit.com/pmsa003i/python-circuitpython

FoamyGuy commented:
I found another learn guide that will also need a small update when this PR gets merged. https://learn.adafruit.com/diy-air-quality-monitor/code-usage

@kattni and @brentru If this is ready to merge, I can work on updating the learn guides and their code if you want?

kattni commented:
@FoamyGuy Please merge this and release the library, and then proceed onto updating the guides. That would be greatly appreciated - thank you!
